### ISSUE TITLE
Automatic deselection bugs

### ISSUE THREAD

**Issue by Mordax** [FlashpointProject/launcher#20]
Refers to #17 -
* [ ] Clicking a platform link does not deselect the current playlist
* [ ] Clicking the Favorites Playlist link does not deselect clear the current search (the Hall of Fame button has the same bug)

**Comment by TBubba** [comment]
- **Deselect current playlist**
  Calling ``onSelectPlaylist`` with the playlist argument set to ``undefined`` will deselect the current playlist. Call it inside ``onPlatformClick`` (don't forget to get the default library route).
- **Clear current search**
  HomePage has a prop called ``clearSearch`` that clears the search when called. Call it right before or after setting the playlist.

**About finding the library route of a playlist:**
Because the number of places that searches for the library path of a playlist (inside HomePage), I think its appropriate to to move it into its own function. I wrote one to save you the trouble. Just copy & paste it at the bottom of ``HomePage.ts`` with the other functions.

```typescript
/**
 * Get the library route of a playlist.
 * Note: A playlist with an empty or missing library route (undefined or '') means that it belongs to the default library.
 * @param playlist Playlist to get the library route of.
 * @param libraries Library collection to search for the default library in.
 * @returns If the playlist has a library route declared, it is returned.
 *          If not, the default library's route is returned instead (if it is found).
 *          If the default library is not found, return undefined.
 */
function getPlaylistLibraryRoute(playlist: IGamePlaylist, libraries: IGameLibraryFileItem[]): string | undefined {
  if (playlist.library) {
    return playlist.library;
  } else {
    const defLibrary = findDefaultLibrary(libraries);
    if (defLibrary) { return defLibrary.route; }
  }
}
```

With this function the on click listeners could be cleaned up to look something like this:
```typescript
  private onHallOfFameClick = () => {
    const { central, clearSearch, libraryData, onSelectPlaylist } = this.props;
    const hof = findHallOfFamePlaylist(central.playlists.playlists);
    const route = hof && getPlaylistLibraryRoute(hof, libraryData.libraries);
    onSelectPlaylist(hof, route);
    clearSearch();
  }
```

**Comment by TBubba** [comment]
Fixed! a09701866d4ef13444febdd775d616bdda8dd190
